src/main/java/fi/otavanopisto/kuntaapi/server/integrations/ptv/translation/PtvOutPtvInTranslator.java

```diff
@@ -396,22 +498,97 @@ private V7VmOpenApiAddressIn translateAddress(V7VmOpenApiAddress address) {
     PtvAddressSubtype subtype = getAddressSubtype(result.getSubType());
     switch (subtype) {
       case POST_OFFICE_BOX:
-        result.setPostOfficeBoxAddress(result.getPostOfficeBoxAddress());
+        result.setPostOfficeBoxAddress(translateAddressPostOfficeBox(address.getPostOfficeBoxAddress()));
       break;
       case NO_ADDRESS:
       break;
       case SINGLE:
       case STREET:
-        result.setStreetAddress(result.getStreetAddress());
+        result.setStreetAddress(translateAddressStreetWithCoordinates(address.getStreetAddress()));
       break;
       default:
-        logger.log(Level.SEVERE, () -> String.format("Unknown subtype %s", result.getSubType()));
+        logger.log(Level.SEVERE, () -> String.format(UNKNOWN_SUBTYPE, result.getSubType()));
       break;
     }
     
     return result;
   }
 
+  private VmOpenApiAddressStreetIn translateAddressStreet(VmOpenApiAddressStreet streetAddress) {
+    if (streetAddress == null) {
+      return null;
+    }
+    
+    VmOpenApiAddressStreetIn result = new VmOpenApiAddressStreetIn();
+    result.setAdditionalInformation(streetAddress.getAdditionalInformation());
+    result.setMunicipality(streetAddress.getMunicipality() != null ? streetAddress.getMunicipality().getCode() : null);
+    result.setPostalCode(streetAddress.getPostalCode());
+    result.setStreet(streetAddress.getStreet());
+    result.setStreetNumber(streetAddress.getStreetNumber());
+    
+    return result;
+  }
+  
+  private VmOpenApiAddressStreetWithCoordinatesIn translateAddressStreetWithCoordinates(VmOpenApiAddressStreetWithCoordinates streetAddress) {
+    if (streetAddress == null) {
+      return null;
+    }
+    
+    VmOpenApiAddressStreetWithCoordinatesIn result = new VmOpenApiAddressStreetWithCoordinatesIn();
+    result.setAdditionalInformation(streetAddress.getAdditionalInformation());
+    result.setLatitude(streetAddress.getLatitude());
+    result.setLongitude(streetAddress.getLongitude());
+    result.setMunicipality(streetAddress.getMunicipality() != null ? streetAddress.getMunicipality().getCode() : null);
+    result.setPostalCode(streetAddress.getPostalCode());
+    result.setStreet(streetAddress.getStreet());
+    result.setStreetNumber(streetAddress.getStreetNumber());
+    
+    return result;
+  }
+  
+  private VmOpenApiAddressStreetWithCoordinatesIn translateAddressStreetWithCoordinates(VmOpenApiAddressStreetWithOrder streetAddress) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused private "translateAddressStreetWithCoordinates" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AUnusedPrivateMethod)